src/transformers/models/mistral/modeling_mistral.py

```diff
@@ -891,11 +890,8 @@ def _update_causal_mask(
         dtype, device = input_tensor.dtype, input_tensor.device
         min_dtype = torch.finfo(dtype).min
         sequence_length = input_tensor.shape[1]
-        # SlidingWindowCache
-        if using_sliding_window_cache:
-            target_length = max(sequence_length, self.config.sliding_window)
-        # StaticCache
-        elif using_static_cache:
+        # SlidingWindowCache or StaticCache
+        if past_key_values is not None and (past_key_values.is_sliding or past_key_values.is_static):
             target_length = past_key_values.get_max_length()
```

Comment:
Because of the changes on the sliding window cache, `target_length` can now be `None`, which means that `causal_mask = torch.full((sequence_length, target_length), ...)` will fail, a few lines below -- right?

I agree that the sliding window has infinite length. To untangle this, I have the following suggestion:
1. Let's rename `get_max_length` to `get_cache_shape` or a similar name. The idea is to have a function to retrieve the shape of the cache object, which is used in other places like here. (`get_max_length`  would have to go through a deprecation cycle, calling `get_cache_shape` under the hood until it is deleted )
2. Explore the current uses of `get_max_length()`. When we want to obtain the shape of the object, we would call `get_cache_shape`. When we want to obtain the maximum cache length, we would do something like `max_length = None if cache.is_sliding else cache.get_max_length()`

WDYT?

Reply:
Oh missed the moment we got rid of `max-length` in sliding window. Sounds good but I am not sure if there are cases when we want to know the `max_length` and not `cache_length`. Oke, let me go through the repo and check all possible cases :)

Reply:
> but I am not sure if there are cases when we want to know the max_length and not cache_length

I think the same :D but let's double-check 🙏